`SpringDataRepositoryCreator` generates duplicate `findAll()` method when extending bare `Repository` marker interface with explicit `findAll()` declaration

### Describe the bug

In the process of updating dependencies in a local branch of a Docker service, I have been unable to build. After some investigation, it appears that the problem occurs when I update Quarkus from `3.35.1` to `3.36.0`.

When a Spring Data repository extends the bare `Repository<T, ID>` marker interface and explicitly declares a `findAll()` method returning `List<T>`, the Quarkus build fails with a Duplicate method added exception.
This is a legitimate and documented Spring Data pattern for creating minimal repositories that expose only specific methods. It works correctly in standard Spring Boot.

### `pom.xml`
```
<properties>
...
<quarkus.platform.version>3.36.0</quarkus.platform.version>
...
</properties>
```

### Expected behavior

The Quarkus container should build successfully, with only one `findAll()` implementation being created

### Actual behavior

The build fails with exit code `1`, producing:

```
Build step io.quarkus.spring.data.deployment.SpringDataJPAProcessor#build threw an exception:
java.lang.IllegalArgumentException: Duplicate method added:
ClassMethod[...#findAll()Ljava/util/List;]
```

### How to Reproduce?

Define a new `Repository` interface. The following code is adapted from my work, though anonymised.

```
import jakarta.enterprise.context.RequestScoped;
import org.springframework.data.repository.Repository;

import java.util.List;

@RequestScoped
public interface ItemRepo extends Repository<Item, Long> {
    List<Item> findAll();
}
```
---

Build the container in Docker

---

You should get a stack trace with exit code `1`

### Output of `uname -a` or `ver`

Linux NDLT1994 6.6.87.2-microsoft-standard-WSL2 #1 SMP PREEMPT_DYNAMIC Thu Jun  5 18:30:46 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "25.0.3" 2026-04-21 LTS

### Quarkus version or git rev

3.36.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

<img width="2000" height="1062" alt="Image" src="https://github.com/user-attachments/assets/afa7b774-607f-4654-ba09-8d26747368e6" />

@geoand we see the similar error in Quarkus Test Suite: https://github.com/quarkus-qe/quarkus-test-suite/
running ` mvn clean verify -pl spring/spring-data/` leads to this failure:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.999-SNAPSHOT:build (build) on project spring-data: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.spring.data.rest.deployment.SpringDataRestProcessor#registerRepositories threw an exception: java.lang.IllegalArgumentException: Duplicate method added: ClassMethod[io.quarkus.ts.spring.data.rest.MagazineListCrudRepositoryResourceImpl_401be78bfc3a1569019e769729ecb2f5b4ac70ed#list(Lio/quarkus/panache/common/Page;Lio/quarkus/panache/common/Sort;Ljava/lang/String;Ljava/util/Map;)Ljava/util/List;]
[ERROR] 	at io.quarkus.gizmo2.impl.ClassCreatorImpl.method(ClassCreatorImpl.java:79)
[ERROR] 	at io.quarkus.spring.data.rest.deployment.RepositoryMethodsImplementor.implementList(RepositoryMethodsImplementor.java:99)
[ERROR] 	at io.quarkus.spring.data.rest.deployment.ResourceImplementor.lambda$implement$0(ResourceImplementor.java:35)
[ERROR] 	at io.quarkus.gizmo2.impl.GizmoImpl.lambda$class_$1(GizmoImpl.java:134)
[ERROR] 	at io.smallrye.classfile.impl.ClassFileImpl.build(ClassFileImpl.java:144)
[ERROR] 	at io.smallrye.classfile.ClassFile.build(ClassFile.java:558)
[ERROR] 	at io.quarkus.gizmo2.impl.GizmoImpl.class_(GizmoImpl.java:131)
[ERROR] 	at io.quarkus.gizmo2.Gizmo.class_(Gizmo.java:89)
[ERROR] 	at io.quarkus.spring.data.rest.deployment.ResourceImplementor.implement(ResourceImplementor.java:30)
[ERROR] 	at io.quarkus.spring.data.rest.deployment.SpringDataRestProcessor.implementResources(SpringDataRestProcessor.java:129)
[ERROR] 	at io.quarkus.spring.data.rest.deployment.SpringDataRestProcessor.registerRepositories(SpringDataRestProcessor.java:98)
[ERROR] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:811)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:242)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1474)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
[ERROR] 

```



UPD: Quarkus commit is 15e9a19722ae36625bf5cb20178c5d04af5535bc (tip of 3.x at the time of writing)

--- a/extensions/spring-data-rest/deployment/src/main/java/io/quarkus/spring/data/rest/deployment/RepositoryMethodsImplementor.java
+++ b/extensions/spring-data-rest/deployment/src/main/java/io/quarkus/spring/data/rest/deployment/RepositoryMethodsImplementor.java
@@ -75,7 +75,8 @@ public class RepositoryMethodsImplementor implements ResourceMethodsImplementor 
     //    CrudRepository Iterable<T> findAll();
     public void implementIterable(ClassCreator classCreator, String repositoryInterfaceName) {
         if (entityClassHelper.isCrudRepository(repositoryInterfaceName)
-                && !entityClassHelper.isPagingAndSortingRepository(repositoryInterfaceName)) {
+                && !entityClassHelper.isPagingAndSortingRepository(repositoryInterfaceName)
+                && !entityClassHelper.isListCrudRepository(repositoryInterfaceName)) {
             classCreator.method("list", mc -> {
                 mc.public_();
                 mc.returning(List.class);

--- a/extensions/spring-data-rest/deployment/src/test/java/io/quarkus/spring/data/rest/crud/DefaultListCrudRecordsRepository.java
+++ b/extensions/spring-data-rest/deployment/src/test/java/io/quarkus/spring/data/rest/crud/DefaultListCrudRecordsRepository.java
@@ -1,0 +1,6 @@
+package io.quarkus.spring.data.rest.crud;
+
+import org.springframework.data.repository.ListCrudRepository;
+
+public interface DefaultListCrudRecordsRepository extends ListCrudRepository<Record, Long> {
+}

--- a/extensions/spring-data-rest/deployment/src/test/java/io/quarkus/spring/data/rest/crud/DefaultListCrudResourceTest.java
+++ b/extensions/spring-data-rest/deployment/src/test/java/io/quarkus/spring/data/rest/crud/DefaultListCrudResourceTest.java
@@ -1,0 +1,39 @@
+package io.quarkus.spring.data.rest.crud;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.spring.data.rest.AbstractEntity;
+import io.quarkus.test.QuarkusExtensionTest;
+
+class DefaultListCrudResourceTest {
+    @RegisterExtension
+    static final QuarkusExtensionTest TEST = new QuarkusExtensionTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(AbstractEntity.class, Record.class, DefaultListCrudRecordsRepository.class)
+                    .addAsResource("application.properties")
+                    .addAsResource("import.sql"));
+
+    @Test
+    void shouldGet() {
+        given().accept("application/json")
+                .when().get("/default-list-crud-records/1")
+                .then().statusCode(200)
+                .and().body("id", is(equalTo(1)))
+                .and().body("name", is(equalTo("first")));
+    }
+
+    @Test
+    void shouldList() {
+        given().accept("application/json")
+                .when().get("/default-list-crud-records")
+                .then().statusCode(200)
+                .and().body("id", hasItems(1, 2))
+                .and().body("name", hasItems("first", "second"));
+    }
+}

